### PR TITLE
Add New Networks

### LINKED ISSUE #1249: Add new networks. Ensure HW wallets are only for certain networks.
- [ ] UBQ
    - brand-ubq: #00ad85;
    - Ledger: True
    - Trezor: True
    - Digital Bitbox: False
    - m/44'/108'/0'/0
- [ ] POA
    - brand-poa: #4568bb;
    - Ledger: True
    - Trezor: True
    - Digital Bitbox: False
    - m/44'/1'/0'/0
- [ ] TOMO
    - brand-tomo: #6a488d;
    - Ledger: True
    - Trezor: True
    - Digital Bitbox: False
     - m/44'/1'/0'/0
- [ ] ELLA
    - brand-ella: #046111;
    - Ledger: False
    - Trezor: True
    - Digital Bitbox: False
     - m/44'/163'/0'/0 : Ellaism
- [ ] ETSC
    - brand-etsc: #673ab7;
    - Ledger: False
    - Trezor: False
    - Digital Bitbox: False

https://github.com/MyCryptoHQ/mycrypto.com/blob/b9773a865a9acbd9749f0b681d937dd83c51ebcf/app/scripts/nodes.js#L170-L234

### PR DESCRIPTION
**Pull request by wbobeirne**
Closes #1249.

### Description

Adds the new networks we were missing from mycrypto.com. Also handles the possibility of some dPaths not being provided for unsupported wallets. **These changes have not been tested significantly, as I don't have a lot of knowledge about these networks and don't hold any of their tokens.**

### Changes

* Added new networks to staticNetworks reducer
* Added nodes for each network in staticNodes reducer
* Added handling for LedgerNano and Trezor wallet unlock components, in the should-be-impossible scenario that they're rendered on an unsupported network.

### Screenshots

screen shot 2018-03-05 at 7 00 27 pm

### CODE REVIEW

**Comment by wbobeirne** [comment]
Fixed unit tests. The entire list of nodes was simply copied in the tests, and I didn't update that. Rather than copy-pasting again, I've removed the copied list as a test as that's a bit useless.

**Comment by dternyak** [comment]
We'll need to have extremely thorough testing of these networks before merge - along with as much type safety for dPaths as possible (there was some issues in the past with dPaths, @HenryNguyen5 should have more information)

File: common/reducers/config/networks/staticNetworks.ts [hunk 1]
```diff
@@ -171,6 +175,83 @@ export const INITIAL_STATE: State = {
...
+      initial: 20
+    }
+  },
+  ETSC: {
+    name: 'ETSC',
+    unit: 'ETSC',
+    chainId: 214,
+    isCustom: false,
+    color: '#673ab7',
+    blockExplorer: makeExplorer('Ethereum Social Explorer', 'https://explorer.ethereumsocial.kr/#'),
```
Comment (akx20000a) [inline common/reducers/config/networks/staticNetworks.ts]: https://explorer.ethereumsocial.kr/# to https://explorer.ethereumsocial.kr/

File: common/reducers/config/networks/staticNetworks.ts [hunk 2]
```diff
@@ -171,6 +175,83 @@ export const INITIAL_STATE: State = {
...
+      max: 60,
+      initial: 20
+    }
+  },
+  ETSC: {
+    name: 'ETSC',
+    unit: 'ETSC',
+    chainId: 214,
+    isCustom: false,
+    color: '#673ab7',
```
Comment (akx20000a) [inline common/reducers/config/networks/staticNetworks.ts]: 4295d1

File: common/reducers/config/networks/staticNetworks.ts [hunk 3]
```diff
@@ -171,6 +175,83 @@ export const INITIAL_STATE: State = {
...
+    contracts: [],
+    dPathFormats: {
+      [SecureWalletName.TREZOR]: TOMO_DEFAULT,
+      [SecureWalletName.LEDGER_NANO_S]: TOMO_DEFAULT,
+      [InsecureWalletName.MNEMONIC_PHRASE]: TOMO_DEFAULT
+    },
+    gasPriceSettings: {
+      min: 1,
+      max: 60,
+      initial: 20
```
Comment (rstormsf) [inline common/reducers/config/networks/staticNetworks.ts]: 1 gwei is preferred

File: common/reducers/config/networks/staticNetworks.ts [hunk 4]
```diff
@@ -171,6 +175,83 @@ export const INITIAL_STATE: State = {
...
+    contracts: [],
+    dPathFormats: {
+      [SecureWalletName.TREZOR]: POA_DEFAULT,
+      [SecureWalletName.LEDGER_NANO_S]: POA_DEFAULT,
+      [InsecureWalletName.MNEMONIC_PHRASE]: POA_DEFAULT
+    },
+    gasPriceSettings: {
+      min: 1,
+      max: 60,
+      initial: 20
```
Comment (rstormsf) [inline common/reducers/config/networks/staticNetworks.ts]: 1 gwei is preferred.

File: common/reducers/config/networks/staticNetworks.ts [hunk 5]
```diff
@@ -171,6 +175,83 @@ export const INITIAL_STATE: State = {
...
       initial: 2
     }
+  },
+  POA: {
+    name: 'POA',
+    unit: 'POA',
+    chainId: 99,
+    isCustom: false,
+    color: '#4568bb',
+    blockExplorer: makeExplorer('Etherchain Light', 'https://core-explorer.poa.network'),
```
Comment (rstormsf) [inline common/reducers/config/networks/staticNetworks.ts]: please use our explorer:
https://poaexplorer.com

File: common/config/dpaths.ts [hunk 6]
```diff
@@ -38,6 +38,26 @@ export const UBQ_DEFAULT: DPath = {
   value: "m/44'/108'/0'/0"
 };
 
+export const POA_DEFAULT: DPath = {
+  label: 'Default (POA)',
+  value: "m/44'/1'/0'/0"
```
Comment (rstormsf) [inline common/config/dpaths.ts]: we used the same as ETH on MEW
```
value: "m/44'/60'/0'/0"
```

File: common/reducers/config/networks/staticNetworks.ts [hunk 7]
```diff
@@ -171,6 +175,83 @@ export const INITIAL_STATE: State = {
       max: 20,
       initial: 2
     }
+  },
+  POA: {
+    name: 'POA',
+    unit: 'POA',
+    chainId: 99,
+    isCustom: false,
+    color: '#4568bb',
```
Comment (rstormsf) [inline common/reducers/config/networks/staticNetworks.ts]: `#6d2eae`

**Comment by rstormsf** [comment]
https://github.com/MyCryptoHQ/MyCrypto/pull/1259/files#diff-7263a21def92c6d328fbff2295a4c3e6R43 
The path we used is the following same as ETH:
```
 value: "m/44'/60'/0'/0"
```

https://github.com/MyCryptoHQ/MyCrypto/pull/1259/files#diff-8d7dfae9c67c4590e78c0308563d8d77R185
For explorer, we use:
[https://poaexplorer.com](https://poaexplorer.com)

https://github.com/MyCryptoHQ/MyCrypto/pull/1259/files#diff-8d7dfae9c67c4590e78c0308563d8d77R196
initial gasPrice: 1 gwei

**Comment by wbobeirne** [comment]
@rstormsf made the requested changes to POA, thanks for taking a look

@akx20000a I've temporarily removed ETSC in this PR because the node is down, and there was some uncertainty about the chain ID change. @dternyak sent you an email to follow up on this, and we can figure it out off of GitHub.

I think this should be good to go now.

**Comment by dternyak** [comment]
Looks great aside from one test failure.

**Comment by wbobeirne** [comment]
Gah, snapshots. Updated and should be passing now.

**Pull request merged**